cassdegrees/templates/widgets/courselistingwidget.html

```diff
@@ -0,0 +1,174 @@
+{% load static %}
+
+{# Usage: Must pass in a string app_name that is id of the particular widget the app is to be bound to#}
+{# Must include imports below once in the relevant template prior to insertion of a widget #}
+{#    <script src="https://cdn.jsdelivr.net/npm/vue/dist/vue.js"></script>#}
+{#    <script src="https://unpkg.com/vue-multiselect@2.1.0"></script>#}
+{#    <script src="https://cdn.jsdelivr.net/npm/vue-resource@1.5.1"></script>#}
+{#    <link rel="stylesheet" href="https://unpkg.com/vue-multiselect@2.1.0/dist/vue-multiselect.min.css">#}
+
+{#    content for multiselect#}
+{#    sources (including for HTML and CSS):#}
+{#    https://vue-multiselect.js.org/#sub-getting-started#}
+{#    https://medium.com/@hugodesigns/how-to-use-the-most-complete-selecting-solution-for-vue-js-f991b2605364#}
+
+{# Be aware that the control has a default maximum options listing of 1000, will need to change it depending on #}
+{# course list sizes #}
+
+<h4>Course listing</h4>
+<div id={{ app_name }}>
+    <multiselect
+            :options="options"
+            :value="optionsProxy"
+            @input="updateSelected"
+            :show-labels="true"
+            :multiple="true"
+            :searchable="true"
+            :close-on-select="false"
+            placeholder="Search courses, press esc or tab to close when done"
+            :custom-label="customLabel"
+            track-by="name"
+            :max-height="300"
+            :open-direction="'top'"
+            :loading="showLoadingSpinner"
+            ref="multiselectref"
+            >
+    </multiselect>
+
+    <ul class="resources-list">
+        <template v-for="(resource, index) in selectedResources">
+            <li class="resource-item" :data-index="index">
+                <div class="resource-info">
+                    <div class="resource-title" :id="index">
+                        <span>[[ resource.code ]] - [[ resource.name ]] - [[resource.id]]</span>
+                    </div>
+                </div>
+                <div class="delete-controls" v-on:click.prevent="removeDependency(index)">
+                    <i class="fa fa-times fa-fw"></i>
+                </div>
+            </li>
+        </template>
+    </ul>
+
+</div>
+
+<script type="application/javascript">
+
```

Comment:
This is likely too long to include inline - can this be moved to a .js file?

Reply:
The benefit, (I think), of including it in a widget template is that the template calling the widget can dynamically pass it new app names (e.g. if multiple entry fields are required on the same page), and the widget template then creates a new vue object for each. I haven't found a way to be able to replicate this separating the file into js. However, all that needs to be included in a calling template is {% include ...widget... app_name %}, so this code is only located in the widget file. Happy to discuss further if you have any ideas?

Reply:
Ah, yep, didn't see that. Thats totally fine then.